intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -0,0 +1,187 @@
+        var chosenFile: String? = null
+        var configFiles: HashSet<String>? = null
+    }
+}
+
+// Based on virtual file events, we update our configs since indexes are always not
+class MirrordConfigWatcher : AsyncFileListener {
+
+    private val adderFileVisitor = object : VirtualFileVisitor<Any?>() {
+        override fun visitFile(file: VirtualFile): Boolean {
+            if (!file.isDirectory && file.path.endsWith("mirrord.json")) {
+                synchronized(this) {
+                    MirrordConfigDropDown.configFiles?.add(file.path)
+                }
+            }
+            return true
+        }
+    }
+
+    private val removerFileVisitor = object : VirtualFileVisitor<Any?>() {
+        override fun visitFile(file: VirtualFile): Boolean {
+            if (!file.isDirectory && file.path.endsWith("mirrord.json")) {
+                synchronized(this) {
+                    MirrordConfigDropDown.configFiles?.remove(file.path)
+                }
+            }
+            return true
+        }
+    }
+
+
+    override fun prepareChange(events: MutableList<out VFileEvent>): AsyncFileListener.ChangeApplier {
+        return object : AsyncFileListener.ChangeApplier {
+            override fun beforeVfsChange() {
+                events.forEach { event ->
+                    when (event) {
+                        is VFileDeleteEvent, is VFileMoveEvent -> {
+                            event.file?.let {
+                                visitChildrenRecursively(it, removerFileVisitor)
+                            }
+                        }
+                    }
+                }
+            }
+
+            override fun afterVfsChange() {
+                events.forEach { event ->
+                    when (event) {
+                        is VFileCreateEvent, is VFileMoveEvent -> {
+                            event.file?.let { file -> visitChildrenRecursively(file, adderFileVisitor) }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// An index for mirrord config files, mapping filePath -> Void
+class MirrordConfigIndex : ScalarIndexExtension<String>() {
+
+    companion object {
+        val key = ID.create<String, Void>("mirrordConfig")
+    }
+
+    override fun getName(): ID<String, Void> {
+        return key
+    }
+
+    override fun getIndexer(): DataIndexer<String, Void, FileContent> {
+        return DataIndexer {
+            val path = it.file.path
+            Collections.singletonMap<String, Void>(path, null)
+        }
+    }
+
+    override fun getKeyDescriptor(): KeyDescriptor<String> {
+        return EnumeratorStringDescriptor.INSTANCE
+    }
+
+    override fun getVersion(): Int {
+        return 0
+    }
+
+    override fun getInputFilter(): FileBasedIndex.InputFilter {
+        return FileBasedIndex.InputFilter {
+            // TODO: replace with a proper regular/glob expression => "glob:?(*.)mirrord.+(toml|json|y?(a)ml)"
```

Comment:
```suggestion
            // TODO: also include toml, yaml and yml extensions once `MirrordConfigApi.getConfigData` supports them.
```

Reply:
im trying to figure out a glob, so it will be resolved

Reply:
Are you also changing `getConfigData` to also parse yaml and toml files? Because otherwise if we match them here, users can use them, and once they try to use them `getConfigData` would try to parse them as json and would fail, wouldn't it?

Reply:
we just do json for now, i have updated